elytron/src/main/java/org/wildfly/extension/elytron/AdvancedModifiableKeyStoreDecorator.java

```diff
@@ -680,6 +721,245 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
     }
 
+    static class ObtainCertificateHandler extends ElytronRuntimeOnlyHandler {
+
+        static final StringListAttributeDefinition DOMAIN_NAMES = new StringListAttributeDefinition.Builder(ElytronDescriptionConstants.DOMAIN_NAMES)
+                .build();
+
+        static final SimpleAttributeDefinition CERTIFICATE_AUTHORITY_ACCOUNT = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY_ACCOUNT, ModelType.STRING, false)
+                .setMinSize(1)
+                .setRestartAllServices()
+                .setCapabilityReference(CERTIFICATE_AUTHORITY_ACCOUNT_CAPABILITY, KEY_STORE_CAPABILITY, true)
+                .build();
+
+        static final SimpleAttributeDefinition AGREE_TO_TERMS_OF_SERVICE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE, ModelType.BOOLEAN, true)
+                .setAllowExpression(true)
+                .build();
+
+        static final SimpleAttributeDefinition STAGING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.STAGING, ModelType.BOOLEAN, true)
+                .setAllowExpression(true)
+                .setDefaultValue(new ModelNode(false))
+                .build();
+
+        static final SimpleAttributeDefinition ALGORITHM = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ALGORITHM, ModelType.STRING, true)
+                .setAllowExpression(true)
+                .setDefaultValue(new ModelNode("RSA"))
+                .setMinSize(1)
+                .build();
+
+        static final SimpleAttributeDefinition KEY_SIZE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.KEY_SIZE, ModelType.INT, true)
+                .setAllowExpression(true)
+                .setDefaultValue(new ModelNode(2048))
+                .setValidator(new IntRangeValidator(1))
```

Comment:
Not directly related to this PR but I wonder if we should have a smarter validator for key size (e.g. only accepts values that are 2^n)

Reply:
I've updated this to use a validator that checks that key size is a power of 2. I've also updated the only other usage of key size (in ```AdvancedModifiableKeyStoreDecorator.GenerateKeyPairHandler```) to use this validator as well.